node/src/tasks/user_submitted_file.rs

```diff
@@ -0,0 +1,111 @@
+use crate::services::file_transfer::commands::FileTransferServiceInterface;
+use crate::tasks::AcceptedBspVolunteer;
+use crate::tasks::StorageHubHandler;
+use crate::tasks::StorageHubHandlerConfig;
+use file_manager::traits::FileStorage;
+use log::{error, info};
+use sc_network::Multiaddr;
+use sc_network::PeerId;
+use sp_core::Blake2Hasher;
+use sp_core::Hasher;
+use std::str::FromStr;
+use storage_hub_infra::event_bus::EventHandler;
+use storage_hub_infra::types::Metadata;
+
+const LOG_TARGET: &str = "user-submitted-file-task";
+
+/// Handles the events related to users submitting files to be stored.
+pub struct UserSubmittedFileTask<SHC: StorageHubHandlerConfig> {
+    storage_hub_handler: StorageHubHandler<SHC>,
+}
+
+impl<SHC: StorageHubHandlerConfig> Clone for UserSubmittedFileTask<SHC> {
+    fn clone(&self) -> Self {
+        Self {
+            storage_hub_handler: self.storage_hub_handler.clone(),
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> UserSubmittedFileTask<SHC> {
+    pub fn new(storage_hub_handler: StorageHubHandler<SHC>) -> Self {
+        Self {
+            storage_hub_handler,
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> EventHandler<AcceptedBspVolunteer>
+    for UserSubmittedFileTask<SHC>
+{
+    async fn handle_event(&self, event: AcceptedBspVolunteer) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "Handling file submitted by user {:?} to BSP with location {:?}",
+            event.owner,
+            event.location,
+        );
+
+        // TODO: use `Multiaddr` instead of `String`.
+        let multiaddresses = event.multiaddresses;
+        let file_metadata = Metadata {
+            owner: event.owner.to_string(),
+            size: event.size.into(),
+            // TODO: use `FileLocation` instead of `String`.
```

Comment:
I see we have `FileLocation`, if we are not missing anything major from it, let's also update it here too.

Reply:
Ok. The `FileLocation<T>` type comes from the runtime, while the `Metadata` type is defined in the `infra` crate. Probably not a good idea to make that crate dependent on the runtime, so I created a new type mirroring the one from the runtime.